### PR TITLE
[Silabs] Add CSL support to silabs platform

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
- PR adds Coordinated Sampled Listening (CSL) support for Silabs Platform
- PR enables SIT ICD features for the Window-App while using the thread synchronized sleepy end device (SSED) feature set

Side effect : Remove MG21 Openthread configs since the board isn't supported anymore.

#### Tests
Manual tests to validate functionality

### CODE REVIEW

File: scripts/examples/gn_silabs_example.sh [hunk 1]
```diff
@@ -192,6 +198,10 @@ else
                 optArgs+="chip_enable_icd_server=true chip_openthread_ftd=false "
                 shift
                 ;;
+            --ssed)
```
Comment (jmartinez-silabs) [inline scripts/examples/gn_silabs_example.sh:201]: do we need the prefix for 1 arg?  or should this enable MTD also?

Comment (mkardous-silabs) [reply scripts/examples/gn_silabs_example.sh:201]: removed

**Review by jmartinez-silabs — APPROVED** [review]

File: examples/window-app/common/window-app.matter [hunk 2]
```diff
@@ -1810,6 +1810,91 @@ cluster UserLabel = 65 {
   readonly attribute int16u clusterRevision = 65533;
 }
 
+/** Allows servers to ensure that listed clients are notified when a server is available for communication. */
+cluster IcdManagement = 70 {
```
Comment (andy31415) [inline examples/window-app/common/window-app.matter:1814]: Is this adding of the ICD management server to the window app related to adding CSL to the silabs platform?

Any chance you could update the summary to also des-abbreviate `CSL` and `SSED` ? what do these stand for?

Comment (mkardous-silabs) [reply examples/window-app/common/window-app.matter:1814]: Updated the description. 
Enabling the SIT ICD feature set isn't related to the SSED feature set, but enabling the SIT ICD feature allows us to show case both together which is fundamentally the end goal.

File: examples/platform/silabs/efr32/project_include/OpenThreadConfig.h [hunk 3]
```diff
@@ -45,35 +45,39 @@
...
 // #ifndef OPENTHREAD_CONFIG_LOG_PREPEND_LEVEL
-// #define OPENTHREAD_CONFIG_LOG_PREPEND_LEVEL         1
+// #define OPENTHREAD_CONFIG_LOG_PREPEND_LEVEL 1
 // #endif
 
-// EFR32MG21A020F1024IM32 has 96k of RAM. Reduce the number of buffers to
-// conserve RAM for this Series 2 part.
-#if defined(EFR32MG21)
-#define OPENTHREAD_CONFIG_NUM_MESSAGE_BUFFERS 22
-#define OPENTHREAD_CONFIG_CLI_UART_TX_BUFFER_SIZE 512
```
Comment (jmeg-sfy) [inline examples/platform/silabs/efr32/project_include/OpenThreadConfig.h:73]: where did it go ?

File: src/platform/silabs/CHIPDevicePlatformConfig.h [hunk 4]
```diff
@@ -118,22 +118,6 @@
 
 // ========== Platform-specific Configuration Overrides =========
 
-#ifndef CHIP_DEVICE_CONFIG_BLE_LL_TASK_PRIORITY
-#define CHIP_DEVICE_CONFIG_BLE_LL_TASK_PRIORITY (configTIMER_TASK_PRIORITY - 1)
-#endif // CHIP_DEVICE_CONFIG_BLE_LL_TASK_PRIORITY
-
-#ifndef CHIP_DEVICE_CONFIG_BLE_STACK_TASK_PRIORITY
-#define CHIP_DEVICE_CONFIG_BLE_STACK_TASK_PRIORITY (CHIP_DEVICE_CONFIG_BLE_LL_TASK_PRIORITY - 1)
```
Comment (jmeg-sfy) [inline src/platform/silabs/CHIPDevicePlatformConfig.h:126]: Where did it go?

**Review by jmeg-sfy — APPROVED** [review]

File: src/platform/silabs/CHIPDevicePlatformConfig.h (hunk 4 above)
Comment (mkardous-silabs) [reply src/platform/silabs/CHIPDevicePlatformConfig.h:126]: same comment

**Pull request merged**